modules/cyusb-tools/src/cygpioset.cpp

```diff
@@ -0,0 +1,344 @@
+//#define ENABLE_DEBUGLOG
+#include <cyusb-tools/cygpio-common.h>
+#include <CyUSBSerial/CyController.hpp>
+#include <CyUSBSerial/CyUSBSerial.hpp>
+#include <vector>
+#include <utility>
+#include <sstream>
+#include <cstring>
+#include <getopt.h>
+
+using OffsetValuePair = std::pair<int, int>;
+using OffsetValuePairArray = std::vector<OffsetValuePair>;
+
+struct AppSettings
+{
+  int gpiochip_num{ -1 };
+  OffsetValuePairArray offset_value_pair_array;
+};
+
+static void print_usage()
+{
+  std::cout << "Usage: cygpioset [OPTIONS] <gpiochip-num> <offset1>=[0|1] [<offset2>=[0|1]]...\n"
+               "\n"
+               "Set GPIO line values of a GPIO chip and maintain the state until the process exits\n"
+               "\n"
+               "Options:\n"
+               "  -h, --help        Display this message and exit\n"
+               //"  -m, --mode=[exit|wait|time|signal] (defaults to 'exit'):\n"
+               //"  -m, --mode=[exit|wait|time] (defaults to 'exit'):\n"
+               //"                    Tell the program what to do after settings values\n"
+               //"  -s, --sec=SEC     Specify the number of seconds to wait (only valid for --mode=time)\n"
+               //"  -u, --usec=USEC   Specify the number of microseconds to wait (only valid for --mode=time)\n"
+               //"\n"
+               //"Modes:\n"
+               //"  exit   Set values and exit immediately\n"
+               //"  wait   Set values and wait for user to press ENTER\n"
+               //"  time   Set values and sleep for a specified amount of time\n"
+               //"  signal:  Set values and wait for SIGINT or SIGTERM\n"
+               "\n"
+               "Example:\n"
+               //"cygpioset -m=time -s=1 gpiochip0 2=0 3=1 4=1 5=0\n"
+               "cygpioset gpiochip0 2=0 3=1 4=1 5=0\n"
+            << std::endl;
+}
+
+static ExitCodeEnum parse_cmdline_args(int argc, char* argv[], AppSettings& r_app_settings_out)
+{
+  ExitCodeEnum enum_exit_code = ExitCodeEnum::SUCCESS;
+
+  int regular_arg_index{};
+
+  static constexpr char a_short_options[] = "-:hn:o:c:";  //"-:aXc:d:";
+
+  int option_index = 0;  // O
+  static struct option a_long_options[] = {
+    { "help", no_argument, nullptr, 'h' },
+    { nullptr, 0, nullptr, 0 }
+  };
+
+  int opt = ::getopt_long(argc, argv, a_short_options, a_long_options, &option_index);
+
+  while (opt != -1)
+  {
+    switch (opt)
+    {
+      case 0:  // long option
+      {
+        const char* p_cstr_long_argument_name = a_long_options[option_index].name;
+        DEBUGLOG("[DEBUG] Option " << p_cstr_long_argument_name << " was provided");
+        if (::optarg == nullptr)
+        {
+          DEBUGLOG(std::endl);
+          /*
+          if (std::strcmp(p_cstr_long_argument_name, "help") == 0)
```

Comment:
no longer needed?

Reply:
This is demonstrative comments on how to parse long form options if they are unique.  I learned later that you can map long options to short options with the 'val' field of the 'option' struct--so in this case, the --help long option will invoke the -h short option handler (opt='h' in the switch/case).  I'm fine with removing this but note if we add long options then we'll have to use this pattern again